app.ts

```diff
@@ -0,0 +1,323 @@
+let totalPages: number;
+let currentPage: number;
+
+const state = {
+	IMQS: "http://localhost:2050",
+	currentValueOfFirstRecord: 1,
+	currentFirstRecordIndex: 1,
+	recordsPerPage: 16,
+	searchedIndex: null as number | null,
+	isButtonDisabled: false
+}
+
+// Fetch the total number of records from the server.
+function totalRecords(): Promise<number> {
+	return fetch(`${state.IMQS}/recordCount`)
+		.then((recordCountResponse) => recordCountResponse.text())
+		.then((recordCountText) => {
+			const recordCount = JSON.parse(recordCountText);
+			return recordCount;
+		}).catch((error) => {
+			throw error;
+		})
+}
+
+// Fetch column names and creates them as table headings
+function fetchColumns() {
+	return fetch(`${state.IMQS}/columns`)
+		.then((columnsResponse) => {
+			return columnsResponse.json();
+		})
+		.then(async (columns) => {
+			const tableHeaderRow = $("#tableHeaderRow");
+			for (const columnName of columns) {
+				const th = document.createElement("th");
+				th.textContent = columnName;
+				tableHeaderRow.append(th);
+			}
+			const recordCount = await totalRecords();
+			totalPages = Math.ceil(recordCount / state.recordsPerPage);
+			updateScreen();
+		}).catch((error) => {
+			throw error;
+		})
+}
+
+// Fetch a range of records from the server based on specified indices.
+function fetchRecords(fromRecord: number, toRecord: number): Promise<any> {
+	if (fromRecord > toRecord) {
+		fromRecord = toRecord
+	}
+	return fetch(`${state.IMQS}/records?from=${fromRecord}&to=${toRecord}`)
+		.then((response) => response.text())
+		.then((dataText) => JSON.parse(dataText))
```

Comment:
You could just say 
```suggestion
		.then((response) => response.json())
```
Instead of first to text and then JSON.parse.
This is applicable to your other fetches as well.
If it is not valid json, the catch will throw your error at least.